src/kindlemint/agents/ai_teammates/architecture_guardian_agent.py

```
@@ -0,0 +1,1135 @@
+
+        for file_obj in all_files:
+            if (
+                file_obj.filename in affected_files
+                and hasattr(file_obj, "patch")
+                and file_obj.patch
+            ):
+                patch_content = file_obj.patch
+
+                # Check for synchronous methods where async is expected
+                if pattern_def.get("description", "").lower().startswith("multi-agent"):
+                    if re.search(r"^[+].*def\s+(?!__)", patch_content, re.MULTILINE):
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)